lib/java/gradle/sourceConfiguration.gradle

```diff
@@ -60,7 +60,16 @@ tasks.withType(JavaCompile) {
     if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
         options.compilerArgs.addAll(['--release', '8'])
     }
-    // options.compilerArgs.addAll('-Xlint:unchecked')
+    options.compilerArgs.addAll([
+            '-Werror',
+            '-Xlint:deprecation',
+            '-Xlint:cast',
+            '-Xlint:empty',
+            '-Xlint:fallthrough',
+            '-Xlint:finally',
+            '-Xlint:overrides',
+            // we can't enable -Xlint:unchecked just yet
```

Comment:
There might be a shorthand for this... something like: `-Xlint:all,-unchecked`
However, I haven't experimented with this enough to know for sure.

Reply:
thanks but i've tried locally, `'-Xlint:all,-unchecked',` does not work as it will still enforce `unchecked`.

in the longer term i'd plan to remove all unchecked code styles so the current explicit enumeration seems fine to me.